File: convert/convert2.py
# 10進数　→　n進数へ変換

n = 18


def convert(n, base):
    result: str = ''

    while n > 0:
        result = str(n % base) + result
        n //= base

    return result


print(convert(n, 2))
print(convert(n, 3))
print(convert(n, 8))

File: convert/convert2.cpp
#include <iostream>
using namespace std;

string result(int n, int base)
{
    string result = "";

    while (n > 0)
    {
        result = to_string(n % base) + result;
        n /= base;
    }

    return result;
}


int main()
{
    int n = 18;

    cout << result(n, 2) << endl;
    cout << result(n, 3) << endl;
    cout << result(n, 8) << endl;

    return 0;
}
